x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/context_menu.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+
+import React, { useState } from 'react';
+import { EuiButtonIcon, EuiContextMenuPanel, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
+
+interface Props {
+  items: React.ReactNode[];
+}
+
+export function ContextMenu({ items }: Props) {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  return (
+    <EuiPopover
+      button={
+        <EuiButtonIcon
+          data-test-subj="infraDashboardsContextMenuButton"
+          display="base"
+          size="s"
+          iconType="boxesVertical"
+          aria-label={i18n.translate('xpack.infra.assetDetails.dashboards.contextMenu.moreLabel', {
+            defaultMessage: 'More',
+          })}
+          onClick={onButtonClick}
+        />
+      }
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      panelPaddingSize="none"
+      anchorPosition="downLeft"
+    >
+      <EuiContextMenuPanel
+        size="s"
+        items={items.map((item: React.ReactNode) => (
```

Comment:
Do you need to type `item`, TS must infer it

```suggestion
        items={items.map((item) => (
```

Reply:
I got this component from APM and didn't check it carefully, I don't think it's needed I'll remove it, thanks 👍